src/main/rx/crsf.c

```diff
@@ -359,15 +572,39 @@ STATIC_UNIT_TESTED uint8_t crsfFrameStatus(rxRuntimeState_t *rxRuntimeState)
 STATIC_UNIT_TESTED float crsfReadRawRC(const rxRuntimeState_t *rxRuntimeState, uint8_t chan)
 {
     UNUSED(rxRuntimeState);
-    /* conversion from RC value to PWM
-     *       RC     PWM
-     * min  172 ->  988us
-     * mid  992 -> 1500us
-     * max 1811 -> 2012us
-     * scale factor = (2012-988) / (1811-172) = 0.62477120195241
-     * offset = 988 - 172 * 0.62477120195241 = 880.53935326418548
-     */
-    return (0.62477120195241f * (float)crsfChannelData[chan]) + 881;
+    if (channelRes == CRSF_RC_RES_CONF_LEGACY) {
+        /* conversion from RC value to PWM
+        * for 0x16 RC frame
+        *       RC     PWM
+        * min  172 ->  988us
+        * mid  992 -> 1500us
+        * max 1811 -> 2012us
+        * scale factor = (2012-988) / (1811-172) = 0.62477120195241
+        * offset = 988 - 172 * 0.62477120195241 = 880.53935326418548
+        */
+        return (0.62477120195241f * (float)crsfChannelData[chan]) + 881;
+    } else {
+        /* conversion from RC value to PWM
+        * for 0x17 Subset RC frame
+        */
+        float scale;
+        switch (channelRes) {
+        case CRSF_SUBSET_RC_RES_CONF_10B:
+            scale = 1.0f;
+            break;
+        default:
+        case CRSF_SUBSET_RC_RES_CONF_11B:
+            scale = 0.5f;
+            break;
+        case CRSF_SUBSET_RC_RES_CONF_12B:
+            scale = 0.25f;
+            break;
+        case CRSF_SUBSET_RC_RES_CONF_13B:
+            scale = 0.125f;
+            break;
+        }
+        return (scale * (float)crsfChannelData[chan]) + 988;
```

Comment:
Could we maybe just use a variable `channelScale` that is set along with `channelBits` and `channelMask` in `crsfFrameStatus`?